packages/core/src/utils.js

```diff
@@ -201,11 +201,21 @@ function computeDefaults(
       )
     );
   } else if ("oneOf" in schema) {
-    schema =
-      schema.oneOf[getMatchingOption(undefined, schema.oneOf, rootSchema)];
+    const matchingKey = getMatchingOption(undefined, schema.oneOf, rootSchema);
+    if (schema.oneOf.hasOwnProperty(matchingKey) === false) {
```

Comment:
Would it be clearer to instead check if `schema.oneOf.length === 0`, so we don't have to unnecessarily call `getMatchingOption`?